template-compose/app/build.gradle.kts

```diff
@@ -160,34 +160,50 @@ dependencies {
     kover(projects.domain)
 }
 
-koverReport {
-    defaults {
-        mergeWith("stagingDebug")
+kover {
+    currentProject {
+        createVariant("custom") {
+            addWithDependencies("stagingDebug")
+        }
+    }
+    reports {
+        // filters for all report types of all build variants
         filters {
-            val excludedFiles = listOf(
-                "*.BuildConfig.*",
-                "*.BuildConfig",
-                // Enum
-                "*.*\$Creator*",
-                // DI
-                "*.di.*",
-                // Hilt
-                "*.*_ComponentTreeDeps*",
-                "*.*_HiltComponents*",
-                "*.*_HiltModules*",
-                "*.*_MembersInjector*",
-                "*.*_Factory*",
-                "*.Hilt_*",
-                "dagger.hilt.internal.*",
-                "hilt_aggregated_deps.*",
-                // Jetpack Compose
-                "*.ComposableSingletons*",
-                "*.*\$*Preview\$*",
-                "*.ui.preview.*",
-            )
-
             excludes {
-                classes(excludedFiles)
+                androidGeneratedClasses()
+                annotatedBy(
+                    // Compose
+                    "androidx.compose.ui.tooling.preview.Preview",
+                    // DI
+                    "dagger.Module",
+                )
```

Comment:
FYI: We found this comment https://github.com/Kotlin/kotlinx-kover/issues/690#issuecomment-2443968341 when updating exclude files on our project, we need to exclude more files to provide accurate code coverage number.